src/DIRAC/ConfigurationSystem/Client/PathFinder.py

```diff
@@ -256,9 +256,20 @@ def useLegacyAdapter(system, service=None) -> bool:
 
     :return: bool -- True if DiracX should be used
     """
+    from DIRAC.Core.Security.ProxyInfo import getProxyInfo
+
+    # Check if DiracX is enabled for this service
     system, service = divideFullName(system, service)
     value = gConfigurationData.extractOptionFromCFG(f"/DiracX/LegacyClientEnabled/{system}/{service}")
-    return (value or "no").lower() in ("y", "yes", "true", "1")
+    isServiceEnabled = (value or "no").lower() in ("y", "yes", "true", "1")
```

Comment:
How about returning quickly and not calling `getProxyInfo` if it's pointless:

```suggestion
    if (value or "no").lower() not in ("y", "yes", "true", "1"):
        return False
```